homeassistant/components/fints/sensor.py

```diff
@@ -115,13 +117,23 @@ class FinTsClient:
     Use this class as Context Manager to get the FinTS3Client object.
     """
 
-    def __init__(self, credentials: BankCredentials, name: str) -> None:
+    def __init__(
+        self,
+        credentials: BankCredentials,
+        name: str,
+        account_config: dict,
+        holdings_config: dict,
+    ) -> None:
         """Initialize a FinTsClient."""
         self._credentials = credentials
+        self._account_information: dict[str, dict] = {}
+        self._account_information_fetched = False
         self.name = name
+        self.account_config = account_config
+        self.holdings_config = holdings_config
 
-    @property
-    def client(self):
+    @cached_property
+    def client(self) -> FinTS3PinTanClient:
         """Get the client object.
 
         As the fints library is stateless, there is not benefit in caching
```

Comment:
The comment mentions that there is no benefit in caching. The property or the comment needs updating.

Reply:
Thanks for pointing this out. The comment refers to the old FinTS library, the current one does store information about bank capabilities and FinTS dialog state, so caching is beneficial.